sonar-scanner-engine/src/main/java/org/sonar/batch/mediumtest/TaskResult.java

```diff
@@ -76,22 +77,22 @@ public void scanTaskCompleted(ProjectScanContainer container) {
     if (!container.getComponentByType(AnalysisMode.class).isIssues()) {
       Metadata readMetadata = getReportReader().readMetadata();
       int rootComponentRef = readMetadata.getRootComponentRef();
-      storeReportComponents(rootComponentRef, null, readMetadata.hasBranch() ? readMetadata.getBranch() : null);
+      storeReportComponents(rootComponentRef, null, readMetadata.getBranch());
     }
 
     storeFs(container);
 
   }
 
-  private void storeReportComponents(int componentRef, String parentModuleKey, @Nullable String branch) {
+  private void storeReportComponents(int componentRef, String parentModuleKey, String branch) {
     Component component = getReportReader().readComponent(componentRef);
-    if (component.hasKey()) {
-      reportComponents.put(component.getKey() + (branch != null ? ":" + branch : ""), component);
+    if (isNotEmpty(component.getKey())) {
+      reportComponents.put(component.getKey() + (isNotEmpty(branch) ? ":" + branch : ""), component);
     } else {
-      reportComponents.put(parentModuleKey + (branch != null ? ":" + branch : "") + ":" + component.getPath(), component);
+      reportComponents.put(parentModuleKey + (isNotEmpty(branch) ? ":" + branch : "") + ":" + component.getPath(), component);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add parentheses to make the operator precedence explicit. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS864)